spec/lib/pact_broker/pacts/repository_spec.rb

```diff
@@ -513,27 +513,74 @@ module Pacts
             .create_consumer("Consumer")
             .create_consumer_version("1.2.2")
             .create_provider("Provider")
+            .create_consumer_version_tag("a_tag")
+            .create_pact
+            .create_consumer_version("1.2.3")
             .create_pact
             .create_consumer_version("1.2.4")
+            .create_consumer_version_tag("another_tag")
             .create_pact
-            .create_consumer_version("1.2.6")
+            .create_consumer_version("1.2.5")
+            .create_consumer_version_tag("a_tag")
+            .create_pact
+            .create_consumer_version("1.2.7")
+            .create_consumer_version_tag("another_tag")
             .create_pact
             .create_provider("Another Provider")
-            .create_consumer_version("1.2.5")
+            .create_consumer_version("1.2.6")
+            .create_consumer_version_tag("a_tag")
             .create_pact
         end
 
-        let(:pact) { Repository.new.find_latest_pact "Consumer", "Provider"  }
+        context "regardless of tag" do
+          context "when a previous version with a pact exists" do
 
-        subject  { Repository.new.find_previous_pact pact }
+            let(:pact) { Repository.new.find_latest_pact "Consumer", "Provider", "another_tag" }
```

Comment:
I try not to use other "business logic" methods to set up my data where possible, so that we reduce the chance of a bug in the other business logic method causing an incorrect failure or pass in the method under test, or a change in the implementation of that method requiring a change to this test. For this reason I tend to use direct Sequel, or the `and_return_xxx` methods from the test data builder. Not a biggy, just my preference.